spec/01-lexical-syntax.md

```diff
@@ -553,6 +553,22 @@ but are required to be properly nested.  Therefore, a comment like
 `/* /* */` will be rejected as having an unterminated
 comment.
 
+## Multi-line and trailing commas
+
+Trailing commas are allowed when followed by a newline and a closing
```

Comment:
Regarding the implementation, I'm not up to speed on the IDE's use of the parser, but it looks related to brace patching:

>  return `true` if subsequent end of line handling should be suppressed.

It's overridden in `scala.tools.nsc.ast.parser.Scanners.UnitScanner#applyBracePatch` according to IntelliJ.